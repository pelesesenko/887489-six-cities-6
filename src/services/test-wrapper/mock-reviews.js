export const mockReviews = [
  {
    id: 1,
    user: {
      id: 18,
      isPro: true,
      name: `Sophie`,
      avatarUrl:
        `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
    },
    rating: 5,
    comment:
      `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
    date: `2021-02-23T08:04:28.647Z`,
  },
  {
    id: 2,
    user: {
      id: 15,
      isPro: false,
      name: `Kendall`,
      avatarUrl:
        `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    },
    rating: 3,
    comment:
      `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2021-02-20T08:04:28.647Z`,
  },
];
