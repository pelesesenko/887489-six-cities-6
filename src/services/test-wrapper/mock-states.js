import {initialState as offers} from "../../store/reducers/offers/offers";
import {initialState as room} from "../../store/reducers/room/room";
import {initialState as favorites} from "../../store/reducers/favorites/favorites";
import {initialState as authorizationStatus} from "../../store/reducers/authorization-status/authorization-status";
import {initialState as sortOrder} from "../../store/reducers/sort-order/sort-order";
import {initialState as currentCityName} from "../../store/reducers/current-city-name/current-city-name";
import {initialState as currentUser} from "../../store/reducers/current-user/current-user";
import {initialState as serverAvailability} from "../../store/reducers/server-availability/server-availability";

export const initial = {
  offers,
  room,
  favorites,
  authorizationStatus,
  sortOrder,
  currentCityName,
  currentUser,
  serverAvailability,
};

export const filled = {
  authorizationStatus: `AUTH`,
  sortOrder: `Popular`,
  currentCityName: `Paris`,
  serverAvailability: true,
  currentUser: {
    id: 1,
    email: `123@mail.ru`,
    name: `123`,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`,
    isPro: false,
  },
  offers: {
    isLoaded: true,
    entities: [
      {
        city: {
          name: `Paris`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13,
          },
        },
        previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
        ],
        title: `Canal View Prinsengracht`,
        isFavorite: false,
        isPremium: true,
        rating: 3.5,
        type: `apartment`,
        bedrooms: 2,
        maxAdults: 7,
        price: 340,
        goods: [
          `Baby seat`,
          `Towels`,
          `Dishwasher`,
          `Coffee machine`,
          `Air conditioning`,
          `Washer`,
          `Fridge`,
          `Breakfast`,
          `Laptop friendly workspace`,
        ],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
        location: {
          latitude: 48.84461,
          longitude: 2.374499,
          zoom: 16,
        },
        id: 1,
      },
      {
        city: {
          name: `Cologne`,
          location: {
            latitude: 50.938361,
            longitude: 6.959974,
            zoom: 13,
          },
        },
        previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        ],
        title: `Nice, cozy, warm big bed apartment`,
        isFavorite: true,
        isPremium: false,
        rating: 2.1,
        type: `room`,
        bedrooms: 1,
        maxAdults: 3,
        price: 196,
        goods: [
          `Laptop friendly workspace`,
          `Fridge`,
          `Breakfast`,
          `Baby seat`,
          `Washer`,
          `Air conditioning`,
          `Towels`,
        ],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
        location: {
          latitude: 50.934361,
          longitude: 6.943974,
          zoom: 16,
        },
        id: 2,
      },
      {
        city: {
          name: `Amsterdam`,
          location: {
            latitude: 52.37454,
            longitude: 4.897976,
            zoom: 13,
          },
        },
        previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
        ],
        title: `Amazing and Extremely Central Flat`,
        isFavorite: false,
        isPremium: false,
        rating: 2.3,
        type: `room`,
        bedrooms: 1,
        maxAdults: 1,
        price: 145,
        goods: [`Laptop friendly workspace`, `Breakfast`],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
        location: {
          latitude: 52.36854,
          longitude: 4.887976,
          zoom: 16,
        },
        id: 3,
      },
      {
        city: {
          name: `Brussels`,
          location: {
            latitude: 50.846557,
            longitude: 4.351697,
            zoom: 13,
          },
        },
        previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
        ],
        title: `Loft Studio in the Central Area`,
        isFavorite: false,
        isPremium: false,
        rating: 3.8,
        type: `apartment`,
        bedrooms: 1,
        maxAdults: 7,
        price: 209,
        goods: [
          `Breakfast`,
          `Air conditioning`,
          `Laptop friendly workspace`,
          `Washer`,
          `Baby seat`,
          `Fridge`,
          `Towels`,
        ],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
        location: {
          latitude: 50.837557,
          longitude: 4.339697,
          zoom: 16,
        },
        id: 4,
      },
      {
        city: {
          name: `Paris`,
          location: {latitude: 48.85661, longitude: 2.351499, zoom: 13},
        },
        previewImage:
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
        ],
        title: `The Joshua Tree House`,
        isFavorite: true,
        isPremium: false,
        rating: 2.1,
        type: `hotel`,
        bedrooms: 4,
        maxAdults: 9,
        price: 346,
        goods: [
          `Dishwasher`,
          `Towels`,
          `Fridge`,
          `Air conditioning`,
          `Baby seat`,
          `Washer`,
          `Laptop friendly workspace`,
          `Breakfast`,
        ],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description:
          `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
        location: {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
        id: 13,
      },
      {
        city: {
          name: `Paris`,
          location: {latitude: 48.85661, longitude: 2.351499, zoom: 13},
        },
        previewImage:
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        ],
        title: `Loft Studio in the Central Area`,
        isFavorite: false,
        isPremium: true,
        rating: 3.4,
        type: `house`,
        bedrooms: 4,
        maxAdults: 7,
        price: 558,
        goods: [
          `Air conditioning`,
          `Breakfast`,
          `Laptop friendly workspace`,
          `Washer`,
          `Baby seat`,
        ],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description:
          `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
        location: {latitude: 48.846610000000005, longitude: 2.374499, zoom: 16},
        id: 23,
      },
      {
        city: {
          name: `Paris`,
          location: {latitude: 48.85661, longitude: 2.351499, zoom: 13},
        },
        previewImage:
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
        images: [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
        ],
        title: `Penthouse, 4-5 rooms + 5 balconies`,
        isFavorite: true,
        isPremium: false,
        rating: 2.6,
        type: `room`,
        bedrooms: 1,
        maxAdults: 2,
        price: 259,
        goods: [
          `Air conditioning`,
          `Breakfast`,
          `Washer`,
          `Laptop friendly workspace`,
        ],
        host: {
          id: 25,
          name: `Angelina`,
          isPro: true,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description:
          `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
        location: {latitude: 48.837610000000005, longitude: 2.364499, zoom: 16},
        id: 76,
      },
    ],
  },
  room: {
    isRoomLoaded: true,
    isNearbyLoaded: true,
    nearIds: [13, 23, 76],
  },
  favorites: {
    isLoaded: true,
    ids: [2, 13, 76],
  },
};

