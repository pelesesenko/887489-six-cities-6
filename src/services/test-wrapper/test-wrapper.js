import React from 'react';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import * as states from './mock-states';

export const history = createMemoryHistory();
export const TestWrapper = ({
  children,
  init = false,
  url = null,
  stateModifier = {},
  mockDispatch = () =>{},
  mockHistory = null,
}) => {

  const mockStore = configureStore({});
  const store = mockStore(() => ({
    ...init ? states.initial : states.filled,
    ...stateModifier,
  }));

  if (!mockHistory) {
    mockHistory = createMemoryHistory();
  }

  if (url) {
    mockHistory.push(url);
  }
  store.dispatch = mockDispatch;

  return (
    <Provider store={store}>
      <Router history={mockHistory}>
        {children}
      </Router>
    </Provider>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.arrayOf(PropTypes.object)]).isRequired,
  init: PropTypes.bool,
  url: PropTypes.string,
  stateModifier: PropTypes.object,
  mockDispatch: PropTypes.func,
  mockHistory: PropTypes.object,
};
