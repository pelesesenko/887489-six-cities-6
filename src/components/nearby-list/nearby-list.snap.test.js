import React from 'react';
import {render} from '@testing-library/react';
import {TestWrapper} from '../../services/test-wrapper/test-wrapper';
import {filled} from '../../services/test-wrapper/mock-states';

import NearbyList from './nearby-list';

it(`NearbyList should render correctly`, () => {

  const nearOffersId = filled.room.nearIds;

  const {container} = render(
      <TestWrapper >
        <NearbyList nearOffersId={nearOffersId}/>
      </TestWrapper>
  );

  expect(container).toMatchSnapshot();
});
