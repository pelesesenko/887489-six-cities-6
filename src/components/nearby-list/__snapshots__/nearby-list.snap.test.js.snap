// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearbyList should render correctly 1`] = `
<div>
  <div
    class="near-places__list places__list"
  >
    <article
      class="place-card near-places__card"
    >
      <div
        class="place-card__image-wrapper near-places__image-wrapper"
      >
        <a
          href="/offer/13"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              346
            </b>
            <span
              class="place-card__price-text"
            >
               / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button place-card__bookmark-button--active"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 40%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/13"
          >
            The Joshua Tree House
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          Hotel
        </p>
      </div>
    </article>
    <article
      class="place-card near-places__card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="place-card__image-wrapper near-places__image-wrapper"
      >
        <a
          href="/offer/23"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              558
            </b>
            <span
              class="place-card__price-text"
            >
               / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button false"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/23"
          >
            Loft Studio in the Central Area
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          House
        </p>
      </div>
    </article>
    <article
      class="place-card near-places__card"
    >
      <div
        class="place-card__image-wrapper near-places__image-wrapper"
      >
        <a
          href="/offer/76"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              259
            </b>
            <span
              class="place-card__price-text"
            >
               / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button place-card__bookmark-button--active"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/76"
          >
            Penthouse, 4-5 rooms + 5 balconies
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          Room
        </p>
      </div>
    </article>
  </div>
</div>
`;
